src/scripts/main.js

```diff
@@ -1,3 +1,22 @@
 'use strict';
 
 // write code here
+const ul = document.querySelector('ul');
+const li = document.querySelectorAll('li');
+
+for (const i of li) {
+  const span = document.createElement('span');
+
+  i.prepend(span);
+  span.prepend(span.nextSibling);
+}
+
+ul.addEventListener('click', (e) => {
+  if (e.target.tagName === 'SPAN') {
+    const targ = e.target.nextSibling;
+
+    if (targ.tagName === 'UL') {
```

Comment:
The condition `if (targ.tagName === 'UL')` may throw an error if `targ` is not an element (e.g., if it's a text node). Consider checking if `targ` is an element before accessing `tagName`.